modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -940,6 +941,88 @@ public void tearDown() throws Exception {
 		_groupLocalService.deleteGroup(_group);
 	}
 
+	@Test
+	public void testDeleteScopeScopeKeyByExternalReferenceCode()
+		throws Exception {
+
+		_siteScopedObjectEntry1 = ObjectEntryTestUtil.addObjectEntry(
+			_siteScopedObjectDefinition1, _OBJECT_FIELD_NAME_1,
+			_OBJECT_FIELD_VALUE_1);
+
+		Group group = _groupLocalService.fetchGroup(
+			TestPropsValues.getGroupId());
+
+		// Using groupExternalReferenceCode
+
+		HTTPTestUtil.invokeToJSONObject(
+			null,
+				StringBundler.concat(
+					_getEndpoint(
+						group.getExternalReferenceCode(), 0,
+						_siteScopedObjectDefinition1),
+					"/by-external-reference-code/",
+					_siteScopedObjectEntry1.getExternalReferenceCode()),
+			Http.Method.DELETE);
+
+		Assert.assertTrue(
+			JSONUtil.isEmpty(
+				HTTPTestUtil.invokeToJSONObject(
+					null,
+					_getEndpoint(
+						null, TestPropsValues.getGroupId(),
+						_siteScopedObjectDefinition1),
+					Http.Method.GET
+				).getJSONArray(
+					"items"
+				)));
+
+		// Using groupId
+
+		HTTPTestUtil.invokeToJSONObject(
+			null,
+			StringBundler.concat(
+				_getEndpoint(
+					null, group.getGroupId(), _siteScopedObjectDefinition1),
+				"/by-external-reference-code/",
+				_siteScopedObjectEntry1.getExternalReferenceCode()),
+			Http.Method.DELETE);
+
+		Assert.assertTrue(
+			JSONUtil.isEmpty(
+				HTTPTestUtil.invokeToJSONObject(
+					null,
+					_getEndpoint(
+						null, TestPropsValues.getGroupId(),
+						_siteScopedObjectDefinition1),
+					Http.Method.GET
+				).getJSONArray(
+					"items"
+				)));
```

Comment:
We should avoid chaining method calls. Have you seen any examples of this pattern in our existing tests?